presto-kafka/src/main/java/io/prestosql/plugin/kafka/KafkaSecurityConfig.java

```diff
@@ -0,0 +1,146 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.prestosql.plugin.kafka;
+
+import io.airlift.configuration.Config;
+import io.airlift.configuration.ConfigDescription;
+import io.airlift.configuration.ConfigSecuritySensitive;
+import io.airlift.configuration.LegacyConfig;
+import io.airlift.configuration.validation.FileExists;
+import org.apache.kafka.common.config.SslConfigs;
+
+import java.util.Properties;
+
+/**
+ *  {@KafkaSecurityConfig} manages Kafka configuration related to connection security. I.e. if the security protocol
+ *  'SSL' is being used. If so, additional properties might be supplied, i.e. location and password of the truststore
+ *  and keystore to be used.
+ */
+public class KafkaSecurityConfig
+{
+    private Properties props;
+
+    private String sslKeystoreFile;
+    private String sslKeystorePassword;
+    private String sslKeyPassword;
+    private String sslTruststoreFile;
+    private String sslTruststorePassword;
+    private KafkaEndpointIdentificationAlgorithm sslEndpointIdentificationAlgorithm;
```

Comment:
set default value to HTTPS

Reply:
I don't forward any property if it wasn't explicitly set via the `kafka.properties` - so as long as `https` is the default in the underlying `kafka-clients` it will  use `https` -> should I set `https` as default anyways?

Reply:
Probably it does not matter but I am not sure how Kafka library reacts if that is set, and plain text communication is used. 
You can filter that out in `getKafkaClientProperties` and only send properties related to SSL if SSL is in use. I think sending it out even if it matches Kafka's default ("https") is fine then.

Reply:
Also talked about this with Praveen - I tried in another kafka-consumer application of mine that uses `kafka-clients`: 
You can supply any ssl related property you want, even if you don't set `security.protocol=SSL` and it is shown (if you let it print out all properties used to instantiate the consumer client) to be used - of course, a configured key store won't/shouldn't have any effect in combination with `security.protocol=PLAINTEXT`...
That's why I simply forward what has been set and do no longer check if `SSL` is set and only then forward further properties (initially had this implemented so that a warning is logged "property XYZ is not used, due to security.protocol=plaintext (default value)"